tests/failpoints/cases/test_disk_full.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use raftstore::store::msg::*;
+use std::time::Duration;
+use std::time::*;
+use test_raftstore::*;
+use tikv_util::config::*;
+/*
+Disk full test will have 3 scenarios.
+[leader full]
+1. not allowed: business wirte, region merge/split,etc.
+2. allowed: read, config change(add/remove peer), transfer leader etc.
+
+[minority full]
+1. not allowed: log entry append
+2. allowed: config change,transfer leader,etc.
+
+[majority full]
+1. equal to minority full, no additional design.
+*/
+const DISK_FULL_LEADER: &str = "disk_full_leader";
+const DISK_FULL_FOLLOWER_1: &str = "disk_full_follower_1";
+
+fn prepare_test_data(cluster: &mut Cluster<ServerCluster>) {
+    let left_key = String::from("5555").into_bytes();
+    let left_value = String::from("55550000").into_bytes();
+    let split_key = String::from("6666").into_bytes();
+    let region = cluster.get_region(b"");
+    cluster.must_split(&region, &split_key);
+    cluster.must_put(&left_key, &left_value);
+    cluster.must_get(&left_key).unwrap();
+    let right_key = String::from("7777").into_bytes();
+    let right_value = String::from("77770000").into_bytes();
+    cluster.must_put(&right_key, &right_value);
+    cluster.must_get(&right_key);
+}
+
+fn fail_leader_full(cluster: &mut Cluster<ServerCluster>) {
+    fail::cfg(DISK_FULL_LEADER, "return").unwrap();
+
+    let leader = cluster.leader_of_region(1).unwrap();
+
+    {
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_1 = raft_stat.get_last_index();
+
+        let key = String::from("8000").into_bytes();
+        let value = String::from("8000").into_bytes();
+        let rx = cluster.async_put(&key, &value).unwrap();
+        is_error_response(&rx.recv_timeout(Duration::from_secs(10)).unwrap());
+
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_2 = raft_stat.get_last_index();
+        assert!(index_1 == index_2);
+    }
+
+    {
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_1 = raft_stat.get_last_index();
+
+        let split_key = String::from("0000").into_bytes();
+        let region = cluster.get_region(&split_key);
+        let split_count_before = cluster.pd_client.get_split_count();
+        cluster.split_region(&region, &split_key, Callback::None);
```

Comment:
Should test the error code in `Callback` instead of comparing region counts before and after the split.

Reply:
done